rules/linux/impact_potential_linux_ransomware_file_encryption.toml

```
@@ -0,0 +1,46 @@
+[metadata]
+creation_date = "2023/03/20"
+integration = ["endpoint"]
+maturity = "production"
+min_stack_comments = "New fields added: required_fields, related_integrations, setup"
+min_stack_version = "8.3.0"
+updated_date = "2023/03/20"
+
+[rule]
+author = ["Elastic"]
+description = """
+Ransomware is a type of malware that encrypts a victim's files or systems and demands payment 
+(usually in cryptocurrency) in exchange for the decryption key. One important indicator of a 
+ransomware attack is the mass encryption of the file system, after which a new file extension 
+is added to the file. This rule identifies a sequence of 50 file extension rename events 
+by the same process in a timespan of 1 second.
+"""
+from = "now-9m"
+index = ["logs-endpoint.events.*"]
+language = "eql"
+license = "Elastic License v2"
+name = "Potential Linux Ransomware Encryption Event Detected"
```

```suggestion
name = "Suspicious File Changes Activity Detected"
```

the keyword `Encryption` means high entropy or random file content which we don't cover with this query